services/headless-lms/models/src/pages.rs

```diff
@@ -2283,6 +2314,81 @@ WHERE pages.order_number = $1
     Ok(())
 }
 
+pub async fn reorder_chapters(
+    conn: &mut PgConnection,
+    chapters: &[Chapter],
+    course_id: Uuid,
+) -> ModelResult<()> {
+    let db_chapters = course_chapters(conn, course_id).await?;
+    let mut tx = conn.begin().await?;
+    // Look for the modified chapter in the existing database
+    for chapter in chapters {
+        if let Some(matching_db_chapter) = db_chapters.iter().find(|c| c.id == chapter.id) {
+            if matching_db_chapter.id == chapter.id {
+                // to avoid conflicting chapter_number when chapter is modified
+                sqlx::query!(
```

Comment:
Move into its own for chapter in chapters loop so that we set the random ones before we make any changes.